list_operations/filtering/lists_filtering.py

```diff
@@ -0,0 +1,64 @@
+import random
+
+import timy
+
+WILDCARD = "*"
+
+
+@timy.timer(loops=1000)
+def filtering_by_filter(objects_to_filter, testing_filters):
+    return list(filter(lambda x: all(
+        [
+            k in x
+            and (v == WILDCARD or x[k] in [v, WILDCARD])
+            for k, v in testing_filters.items()
+        ]
+    ), objects_to_filter))
+
+
+@timy.timer(loops=1000)
+def comprehension_filtering(objects_to_filter, testing_filters):
+    return [ # from Channels._check_filters
+        candidate
+        for candidate in objects_to_filter
+        if all(
+            [
+                k in candidate
+                and (v == WILDCARD or candidate[k] in [v, WILDCARD])
+                for k, v in testing_filters.items()
+            ]
```

Comment:
is it possible not to create this list though (use a generator)

Reply:
```
all(
            
                k in candidate
                and (v == WILDCARD or candidate[k] in [v, WILDCARD])
                for k, v in testing_filters.items()
)
```

Reply:
Good idea, I'll check that